### PR TITLE
Upgrade OTel to version 1.14.0/0.37.0

### PR DESCRIPTION
**Pull request by aaronjheng**
See https://github.com/open-telemetry/opentelemetry-go/releases/tag/v1.14.0.

### CODE REVIEW

File: example/go.sum [hunk 1]
```diff
@@ -1,502 +1,67 @@
-cloud.google.com/go v0.26.0/go.mod h1:aQUYkXzVsufM+DwF1aE+0xfcU+56JwCaLick0ClmMTw=
```
Comment (julienschmidt) [inline example/go.sum:1]: this file was probably deleted / not regenerated by accident?

Comment (aaronjheng) [reply example/go.sum:1]: Nope. I just ran `go mod tidy`.

Comment (julienschmidt) [reply example/go.sum:1]: Maybe not in the `example/` dir? Otherwise there should still be a go.sum 🤔

File: instruments.go [hunk 2]
```diff
@@ -49,7 +48,7 @@ func newInstruments(meter metric.Meter) (*instruments, error) {
 	if instruments.latency, err = meter.Float64Histogram(
 		strings.Join([]string{namespace, "latency"}, "."),
 		instrument.WithDescription("The latency of calls in milliseconds"),
-		instrument.WithUnit(unit.Milliseconds),
```
Comment (XSAM) [inline instruments.go:52]: Any reason to change this?

File: sql_test.go [hunk 3]
```diff
@@ -150,7 +151,8 @@ func TestRegisterDBStatsMetrics(t *testing.T) {
 	assert.NoError(t, err)
 
 	// Should collect 7 metrics
-	got, err := r.Collect(context.Background())
+	got := &metricdata.ResourceMetrics{}
+	err = r.Collect(context.Background(), got)
```
Comment (XSAM) [inline sql_test.go:155]: I thought the original style was quite clear. Any reason to change this?

**Review by XSAM — COMMENTED** [review]
@aaronjheng Thank you for the PR.

Overall good, some comments need to be resolved.

File: instruments.go (hunk 2 above)
Comment (aaronjheng) [reply instruments.go:52]: API changes. See
https://github.com/open-telemetry/opentelemetry-go/pull/3776
https://github.com/open-telemetry/opentelemetry-go/pull/3814

File: sql_test.go (hunk 3 above)
Comment (aaronjheng) [reply sql_test.go:155]: API changes. See 
https://github.com/open-telemetry/opentelemetry-go/pull/3732

**Review by XSAM — APPROVED** [review]

**Comment by XSAM** [comment]
@aaronjheng Thank you for your contribution.

**Comment by natewoods-wf** [comment]
:wave: Thanks for doing the work here @aaronjheng! Any word/timeline on pushing a tag with this @XSAM?

**Comment by XSAM** [comment]
Hi @natewoods-wf, I will make a release later this day.

**Pull request merged**